java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
```

Comment:
Okay, if we do want to return UrlDevices, how about we rename the existing `PwsResult` to `PageMetadata` (or something), and change `getPwPairsSortedByRank` to return a `List<PwResult>` where `PwResult` is:

```
class PwResult {
   PageMetadata page_md;
   List<UrlDevice> devices;
}
```

Then we can always have deduplicated results, and still have access to all the beacon info if we want it.

WDYT?